crypto/merklesignature/persistentMerkleSignatureScheme.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+/* This file contains every database and persistence related method for the merkle Secrets.
+ * It is used when generating the State Proof keys (storing them into a database), and for
+ * importing those keys from the created database file into the algod participation registry.
+ */
+
+package merklesignature
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/util/db"
+)
+
+const merkleSignatureSchemaVersion = 1
+const merkleSignatureTableSchemaName = "merklesignaturescheme"
```

Comment:
The version isn't set on the DB, this logic looks incomplete.

There's a  DB utility to automate versioning & upgrading SQLite DBs, you use `db.Initialize` and pass in functions that create/modify the DB: https://github.com/algorand/go-algorand/blob/c8c6fb7d2395e2298342824eeadba0d498cd3228/util/db/initialize.go#L34

Here is how the participation registry uses it: https://github.com/algorand/go-algorand/blob/c8c6fb7d2395e2298342824eeadba0d498cd3228/data/account/participationRegistry.go#L237